compressible/baroclinic_wave.py

```diff
@@ -0,0 +1,246 @@
+    # Perturbation
+    # -------------------------------------------------------------- #
+
+    base_zonal_u = wind
+    base_merid_u = Constant(0.0)
+
+    a = 6.371229e6
+    zt = 1.5e4     # top of perturbation
+    d0 = a / 6     # horizontal radius of perturbation
+    a = 6.371229e6
+    err_tol = 1e-12
+    lon_c, lat_c = pi/9, 2*pi/9  # location of perturbation centre
+
+    d = a * acos(sin(lat_c)*sin(lat) + cos(lat_c)*cos(lat)*cos(lon - lon_c))  # distance from centre of perturbation
+
+    depth = r - a  # The distance from origin subtracted from earth radius
+    zeta = conditional(ge(depth, zt-err_tol), 0, 1 - 3*(depth / zt)**2 + 2*(depth / zt)**3)  # peturbation vertical taper
+
+    perturb_magnitude = (16*Vp/(3*sqrt(3))) * zeta * sin((pi * d) / (2 * d0)) * cos((pi * d) / (2 * d0))**3
+
+
+    zonal_pert = conditional(le(d, err_tol), 0,
+                             conditional(ge(d, (d0-err_tol)), 0, -perturb_magnitude * (-sin(lat_c)*cos(lat) + cos(lat_c)*sin(lat)*cos(lon - lon_c)) / sin(d / a)))
+    meridional_pert = conditional(le(d, err_tol), 0,
+                                  conditional(ge(d, d0-err_tol), 0, perturb_magnitude * cos(lat_c)*sin(lon - lon_c) / sin(d / a)))
+
+    zonal_u = base_zonal_u + zonal_pert
+    merid_u = base_merid_u + meridional_pert
+    radial_u = Constant(0.0)
+    # Get spherical basis vectors, expressed in terms of thier (x,y,z) components:
+    e_lon = xyz_vector_from_lonlatr(1, 0, 0, xyz)
+    e_lat = xyz_vector_from_lonlatr(0, 1, 0, xyz)
+    e_r = xyz_vector_from_lonlatr(0, 0, 1, xyz)
+
+    # obtain initial conditions
+    print('Set up initial conditions')
+    print('project u')
+    test_u = TestFunction(Vu)
+    dx_reduced = dx(degree=4)
+    u_field = zonal_u*e_lon + merid_u * e_lat + radial_u * e_r
+    u_proj_eqn = inner(test_u, u0 - u_field) * dx_reduced
+    u_proj_prob = NonlinearVariationalProblem(u_proj_eqn, u0)
+    u_proj_solver = NonlinearVariationalSolver(u_proj_prob)
+    u_proj_solver.solve()
+
+    print('interpolate theta')
+    theta0.interpolate(theta_expr)
+    print('find pi')
+    pie = Function(Vr).interpolate(pie_expr)
+    print('find rho')
+    rho0.interpolate(rho_expr)
+    compressible_hydrostatic_balance(eqn, theta0, rho0, exner_boundary=pie, solve_for_rho=True)
+
+    print('make analytic rho')
+    rho_analytic = Function(Vr).interpolate(rho_expr)
+    print('Normalised rho error is:', errornorm(rho_analytic, rho0) / norm(rho_analytic))
+
+    # make mean fields
+    print('make mean fields')
+    rho_b = Function(Vr).assign(rho0)
+    theta_b = Function(Vt).assign(theta0)
+
+    # assign reference profiles
+    stepper.set_reference_profiles([('rho', rho_b),
+                                    ('theta', theta_b)])
+
+    stepper.run(t=0, tmax=tmax)
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser(
+        description="The deep atmosphere dry baroclinic wave Test case from Ullich et al. 2013",
```

Comment:
```suggestion
        description=__doc__,
```
This is what I was trying to remember when I was live coding things!